app/src/main/kotlin/com/hippo/ehviewer/client/EhTagDatabase.kt

```diff
@@ -62,53 +62,37 @@ object EhTagDatabase : CoroutineScope {
 
     fun getTranslation(prefix: String? = NAMESPACE_PREFIX, tag: String?): String? = tagGroups[prefix]?.get(tag)?.trim()?.ifEmpty { null }
 
-    private fun rawSuggestSequence(tags: Map<String, String>, keyword: String, translate: Boolean, exactly: Boolean) = sequence {
-        if (exactly) {
-            tags[keyword]?.let {
-                yield(keyword to it.takeIf { translate })
+    context(Context)
+    fun suggestion(rawKeyword: String, expectTranslate: Boolean): Sequence<Pair<String, String?>> {
+        if (!initialized) return emptySequence()
+        val translate = expectTranslate && isTranslatable(implicit<Context>())
+        val keyword = PREFIXES.fold(rawKeyword) { kwd, pfx -> kwd.removePrefix(pfx) }
+        val prefix = rawKeyword.dropLast(keyword.length)
+        val ns = keyword.substringBefore(':')
+        val tag = keyword.drop(ns.length + 1)
+        val nsPrefix = TagNamespace.from(ns)?.prefix ?: ns
+        val tags = tagGroups[nsPrefix.takeIf { tag.isNotEmpty() && it != NAMESPACE_PREFIX }]
+        fun suggestOnce(exactly: Boolean) = let {
+            fun lookup(tags: Map<String, String>, keyword: String) = when {
+                exactly -> tags[keyword]?.let { t -> sequenceOf(keyword to t.takeIf { translate }) } ?: emptySequence()
+                translate -> tags.asSequence().filter { (tag, hint) ->
+                    tag != keyword && (tag.containsIgnoreSpace(keyword) || hint.containsIgnoreSpace(keyword))
+                }.map { (tag, hint) -> tag to hint }
+                else -> tags.keys.asSequence().filter { tag ->
+                    tag != keyword && tag.containsIgnoreSpace(keyword)
+                }.map { tag -> tag to null }
             }
-        } else {
-            if (translate) {
-                tags.forEach { (tag, hint) ->
-                    if (tag != keyword && (tag.containsIgnoreSpace(keyword) || hint.containsIgnoreSpace(keyword))) {
-                        yield(tag to hint)
-                    }
+            when {
+                tags != null -> lookup(tags, tag).map { (tag, hint) -> "$prefix$nsPrefix:$tag" to hint }
+                nsPrefix != NAMESPACE_PREFIX -> tagGroups.asSequence().flatMap { (nsPrefix, tags) ->
```

Comment:
We want to compare `NAMESPACE_PREFIX` with the prefix of the tags in database, not the input text.